### PR TITLE
Use Pants to test, lint and typecheck in this repo.

### PR DESCRIPTION
**Pull request by benjyw**
We have this neat build system, might as well use it...

### CODE REVIEW

**Comment by benjyw** [comment]
Currently using this to test https://github.com/pantsbuild/actions/pull/16

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -4,33 +4,35 @@ concurrency:
...
-      - name: Shellcheck
-        run: shellcheck pantsup.sh
-      - name: Run Checks
-        uses: pantsbuild/actions/run-tox@v2
-        with:
-          tox-env: format-check,lint,typecheck
-  unit-tests:
+      - name: Lint and Typecheck
+        run: |
+          pants lint check ::
```
Comment (stuhood) [inline .github/workflows/ci.yml]: Nit: I find it more useful to get semantic errors first.
```suggestion
          pants check lint ::
```

File: pants.toml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+[test]
+extra_env_vars = [
+    "PYENV_ROOT",
+    "HOME",
+    "PATH",
+]
+timeout_default = 600
+
+[shellcheck]
+args = ["--external-sources"]
```
Comment (stuhood) [inline pants.toml:31]: This was surprising to me: https://github.com/pantsbuild/pants/issues/17956

**Review by stuhood — APPROVED** [review]

File: pants.toml (hunk 2 above)
Comment (benjyw) [reply pants.toml:31]: Yeah!

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -8,30 +8,17 @@ env:
 jobs:
   org-check:
     name: Check GitHub Organization
-    if: ${{ github.repository_owner == 'pantsbuild' }}
+    #if: ${{ github.repository_owner == 'pantsbuild' }}
```
Comment (stuhood) [inline .github/workflows/ci.yml]: TODO?

**Review by stuhood — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 3 above)
Comment (benjyw) [reply .github/workflows/ci.yml]: Ooops, reverted

**Pull request merged**